spark/src/main/java/org/apache/iceberg/actions/BaseSparkAction.java

```diff
@@ -46,10 +48,11 @@
 
 import static org.apache.iceberg.MetadataTableType.ALL_MANIFESTS;
 
-abstract class BaseSparkAction<ThisT, R> implements Action<ThisT, R> {
+public abstract class BaseSparkAction<ThisT, R> implements Action<ThisT, R> {
```

Comment:
Do we need it to be public?

Reply:
Temporarily yes since we are using it in another package. Once we get rid of the old actions, we should be able to make this non-public.

Reply:
If this is public but will be removed, we should mark it deprecated so people know not to rely on it.

Reply:
This class will be used even after refactoring. It may be moved but to another package, though.